plugins/worldinterface/src/worldproxy.cpp

```diff
@@ -0,0 +1,348 @@
+// -*- mode:C++; tab-width:4; c-basic-offset:4; indent-tabs-mode:nil -*-
+/*
+ * Copyright (C) 2015 iCub Facility 
+ * Authors: Lorenzo Natale
+ * CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ *
+ */
+ 
+ #include "worldproxy.h"
+
+#include <math.h>
+
+#include <string>
+#include <iostream>
+#include <ostream>
+
+using namespace std;
+using namespace gazebo;
+
+#include <yarp/os/Time.h>
+
+void replace(string &str, string key, double value)
+{
+  ostringstream tmp;
+  tmp << value;
+
+  size_t pos = 0;
+    while ((pos = str.find(key, pos)) != string::npos) {
+         str.replace(pos, key.length(), tmp.str());
+         pos += tmp.str().length();
+    }
+}
+
+
+WorldProxy::WorldProxy():
+isSynchro(true)
+{}
+
+WorldProxy::~WorldProxy()
+{}
+
+std::string WorldProxy::makeSphere(const double radius, const GazeboYarpPlugins::Pose& pose, const GazeboYarpPlugins::Color& color)
+{
+  sdf::SDF sphereSDF;
+ 
+  string sphereSDF_string=string(
+      "<?xml version='1.0'?>\
+       <sdf version ='1.4'>\
+          <model name ='sphere'>\
+            <pose>POSEX POSEY POSEZ ROLL PITCH YAW</pose>\
+            <link name ='link'>\
+              <pose>POSEX POSEY POSEZ ROLL PITCH YAW</pose>\
```

Comment:
Are you sure about this duplicated pose ? This will basically mean that the model frame is placed with the requested pose in the world, and then the link frame is placed with the requested pose wrt the model pose. In the end you applied the requested pose two times, I think it make sense to just put the requested pose in the model tag.